### ISSUE TITLE
Use of file_name and file_location [REPLACEMENT ISSUE]

### ISSUE THREAD

**Issue by niconoe** [tdwg/camtrap-dp#18]
The original issue

	Id: 18
	Title: Use of file_name and file_location

could not be created.
This is a dummy issue, replacing the original one. It contains everything but the original issue description. In case the gitlab repository is still existing, visit the following link to show the original issue:

TODO

**Comment by niconoe** [comment]
In GitLab by @kbubnicki on Aug 28, 2020, 14:21

The `file_location` can be a path to a file within a data package or remote URL and does not have to end with a filename i.e. the example `https://trapper.org/storage/resource/media/259024/file/` is correct as this is a proper/accessible URL (this particular one is fake of course).

`file_name` I would keep as this is just a very basic property and it is critically needed when the remote URL does not contain a file name (but it is still fully correct URL).

Then we can consider splitting `file_location` into `file_path` and `file_url`. Maybe this makes sense as you can potentially provide both (i.e. you have your file locally in a data package but it is also accessible online). @all ?

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 28, 2020, 14:27

Can you elaborate on:

> [file_name] is critically needed when the remote URL does not contain a file name

Why is file_name so important?

Also pinging @yliefting: does the user absolutely need to know the name of the file? Or is the path and/or url to access it (which often but not always contain the file name) enough?

**Comment by niconoe** [comment]
In GitLab by @kbubnicki on Aug 28, 2020, 14:46

You are right: `file_name` it is not critically needed when thinking about an enduser working with data (doing analysis, merging datasets etc).

It can be useful for importing data into different platforms. At least, I am sure it is important for importing the existing classifications into TRAPPER (as `deployment_id` and `file_name` give a unique combination in our case). But this is software specific and we can adapt TRAPPER (also see my comment in #10 ;).

In conclusion it is handy but not critical, you are right. It can be removed potentially.

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 28, 2020, 15:07

If useful, we can keep `file_name` around as a non-required term... but maybe we should see how far we get with just removing it.

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 28, 2020, 15:11

Regarding `path` and `url`:

If we look at the data package (resource) specs as a guidance, (data) files are referred to by the property `path` (https://specs.frictionlessdata.io/data-resource/#path-data-in-files) which is of the form `url-or-path`. In old examples, one could use `path` OR `url` (not both): https://frictionlessdata.io/data-package/#resources, but it seems `url` has been deprecated in favour of `path` which can either be local or http.

One issue with using `path` and `url` for files is that you have to define that ONE OF BOTH is required and that both are allowed to be populated. Note sure how to encode that in the schema.

**Comment by niconoe** [comment]
In GitLab by @yliefting on Aug 28, 2020, 15:34

I don't think the filename is absolutely required as long as there is a unique identifier and path to it. That can be a local path or URL, depending on the purpose of the exported data package: analysis (URL) or archival (relative path in package).

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 28, 2020, 15:38

Ok, so @yliefting you would only include one term for path/url, i.e. not supporting the sharing of a local path AND url? That definitely makes it easier to define, I was just wondering if there is a use case for sharing both?

**Comment by niconoe** [comment]
In GitLab by @yliefting on Aug 28, 2020, 15:46

Yes, I would make this dependent on whether or not the assets are part of the package. IF there are no assets, you expect to see an URL and if they are in the package you expect a local path. Having both is ok, but a package with assets meant for archival means the original URL's will most likely stop working at some point. Otherwise you would not archive the data in a repository.

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 28, 2020, 15:56

Ok, I'm curious about @kbubnicki take on this, but it does make things easier and is in line with data package. I would name the field `path` or `file_path` then.

**Comment by niconoe** [comment]
In GitLab by @kbubnicki on Aug 28, 2020, 16:53

Ok! I am fine with one term for path/url which is context-dependent (originally `file_location` was supposed to work exactly the same way). But what about keeping `file_name` as an optional field (for url case)?

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 29, 2020, 15:27

If it helps certain use cases, then I have no objection adding `file_name` as an optional term. We would have to indicate clearly to the user that the full path to the file should be in `file_path`, and that it is not `file_path` + `file_name` = full path.

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 31, 2020, 17:19

mentioned in commit b172b07589fa560eb9245be3007fea3badb3c6db

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Aug 31, 2020, 17:26

mentioned in merge request !4

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Sep 2, 2020, 15:22

Consensus is:

- `file_path`: full path to file as a `url-or-path` (required term). This is an OR situation: either a local path (e.g. archived package) or a full URL (hosted images).
- `file_name`: convenience term with name of file (optional term)

**Comment by niconoe** [comment]
In GitLab by @kbubnicki on Sep 3, 2020, 00:41

mentioned in commit 1cf8d960d276ce21e6b3f2f75e918b241f04545d

**Comment by niconoe** [comment]
In GitLab by @kbubnicki on Sep 3, 2020, 00:46

mentioned in merge request !9

**Comment by niconoe** [comment]
In GitLab by @peterdesmet on Sep 4, 2020, 13:33

Implemented in !9.
